pkg/discovery/repository/kube_cluster.go

```diff
@@ -192,6 +195,67 @@ func (summary *ClusterSummary) computePodToServiceMap() {
 	}
 }
 
+func (summary *ClusterSummary) computeStaticPodToDaemonMap() {
+	glog.V(2).Info("creating mapping for static pods modeled as daemons.")
+	// static pods are only exposed through the API via mirror pods
+	// https://kubernetes.io/docs/reference/glossary/?all=true#term-mirror-pod
+	mirrorPods := util.GetMirrorPods(summary.Pods)
+	// common name prefix is our assumed conventions for showing intent of creating
+	// daemon sets with static pods
+	prefixToNodeNames := util.GetMirrorPodPrefixToNodeNames(mirrorPods)
+	nodePoolToNodeNames := util.MapNodePoolToNodeNames(summary.Nodes)
+	anyPools := len(nodePoolToNodeNames) != 0
+	if anyPools {
+		glog.V(3).Info("Conducting static pod daemon check with node pools.")
+	} else {
+		glog.V(3).Info("Conducting static pod daemon check with nodes.")
```

Comment:
As mentioned, if `MapNodePoolToNodeNames` returns a fake pool to represent the entire cluster, then you don't need the logic of `anyPools`, etc.

Reply:
If creating a fake pool inside `MapNodePoolToNodeNames` is a concern, maybe we can create it here when `antyPools` is false? Then we can simplify the logic below.